engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -197,10 +198,24 @@ public Function lookupMethodDefinition(AtomConstructor atom, String name) {
     if (definedWithAtom != null) {
       return definedWithAtom;
     }
+
     Function definedHere = getMethodMapFor(atom).get(lowerName);
     if (definedHere != null) {
       return definedHere;
     }
+
+    if (atom instanceof BuiltinAtomConstructor) {
```

Comment:
Yeah, so for a fully constructed `BuiltinAtomConstructor`, this always triggers: i.e. there are no methods in the definition scope (because they are all attached via `@Builtin_Method` in the shadow scope). So you could overwrite the def scope and this becomes nicer?